airflow/models/baseoperator.py

```diff
@@ -359,9 +363,12 @@ def __init__(
         do_xcom_push: bool = True,
         inlets: Optional[Any] = None,
         outlets: Optional[Any] = None,
+        task_group: Optional["TaskGroup"] = None,
         **kwargs
     ):
         from airflow.models.dag import DagContext
+        from airflow.utils.task_group import TaskGroupContext
```

Comment:
There is not circular import here. However, I thought putting import inline if it's only needed in this function is better because in the future people might need to import BaseOperator inside task_group.py. Also this is put at this line because I saw `DagContext` being imported here.

Reply:
There is runtime overhead for every import statement even though the module has already been imported. the performance implication is probably not going to matter too much for this particular case, but in general, it's a good coding practice to keep import at the top of the file unless there is no other choice.